### ISSUE TITLE
24h version?

### ISSUE THREAD

**Issue by msebald** [DIY-Machines/DigitalClockSmartShelving#1]
Great DIY project, I will built one myself, too. But is it possible to get a 24h version of this clock as well? Some extra wiring and some different code for the Arduino is needed then. Also the board needs to be extended a bit.

**Comment by lissy93** [comment]
Yeah, I was thinking the same- but the code is not particularly complex, so I think extending it yourself/ myself won't be hard. I'd say this repo is good as it is.

**Comment by sam-tj** [comment]
Yes you can have a 24 hr version.
Just remove that (-12 ) from the hour display code.

**Comment by etnie83** [comment]
I build that clock since 3 weeks... Okay the printer ist working since 3 weeks :-)
I can send a link to my repo If it is finished with 24h-mode and a minus or line between the segments to realise a blink frequenz for the seconds

**Comment by sam-tj** [comment]
> 
> 
> I build that clock since 3 weeks... Okay the printer ist working since 3 weeks :-)
> I can send a link to my repo If it is finished with 24h-mode and a minus or line between the segments to realise a blink frequenz for the seconds

Hi, sorry but I didnt get your question. Is it that you need a section for seconds also?

**Comment by msebald** [comment]
Sorry for answering that late.

@sam-tj I think the bigger problem (for me) is not the code itself but the need for more modules. With a 24 hour version the first digit can not only be '' or '1', it can also be '2'. So another full digit is needed there, too.

@etnie83 A link to your repo would be great, thank you! Do you also have a photo/video? I also do not understand what you did with the seconds. ;-)

**Comment by KatzeMau** [comment]
> I build that clock for 3 weeks... Okay, the printer ist working for 3 weeks :-)
> I can send a link to my repo If it is finished with 24h-mode and a minus or line between the segments to realize a blink frequency for the seconds

Yeah please post the link, I'm making one now it should be finished in 13 days :)
HERE IS THE LINK: https://github.com/etnie83/GiantLEDClock

**Comment by Winston-Lu** [comment]
I made an alternate version that adds a lot more features such as web-server control and a lot of lighting effects, but would require a bit of configuration to get it set up. My setup uses an ESP8266 instead, but has a lot more quality of life features. If anyone has a 24hr version and would like to test this version, I would appreciate any feedback

https://github.com/Winston-Lu/ESP8266-LED-Shelf
